Doc/library/compression.zstd.rst

```diff
@@ -0,0 +1,780 @@
+:mod:`!compression.zstd` --- Compression compatible with the Zstandard format
+=============================================================================
+
+.. module:: compression.zstd
+   :synopsis: Low level interface to compression and decompression routines in
+              Meta's zstd library
+
+.. versionadded:: 3.14
+
+**Source code:** :source:`Lib/compression/zstd/__init__.py`
+
+--------------
+
+This module provides classes and functions for compressing and
+decompressing data using the Zstandard (or *zstd*) compression algorithm. Also
+included is a file interface that supports reading and writing the contents of ``.zst`` files.
+files created by the :program:`zstd` utility, as well as raw zstd compressed
+streams.
+
+The :mod:`compression.zstd` module contains:
+
+* The :func:`.open` function and :class:`ZstdFile` class for reading and
+  writing compressed files.
+* The :class:`ZstdCompressor` and :class:`ZstdDecompressor` classes for
+  incremental (de)compression.
+* The :func:`compress` and :func:`decompress` functions for one-shot
+  (de)compression.
+* The :func:`train_dict` and :func:`finalize_dict` functions and the
+  :class:`ZstdDict` class to train and manage Zstandard dictionaries.
+* The :class:`CompressionParameter`, :class:`DecompressionParameter`, and
+  :class:`Strategy` classes for setting advanced (de)compression parameters.
+
+
+.. exception:: ZstdError
+
+   This exception is raised when an error occurs during compression or
+   decompression, or while initializing the (de)compressor state.
+
+
+Reading and writing compressed files
+------------------------------------
+
+.. function:: open(file, /, mode='rb', *, level=None, options=None, \
+                   zstd_dict=None, encoding=None, errors=None, newline=None)
+
+   Open a Zstandard-compressed file in binary or text mode, returning a
+   :term:`file object`.
+
+   The *file* argument can be either an actual file name (given as a
+   :class:`str`, :class:`bytes` or :term:`path-like <path-like object>` object),
+   in which case the named file is opened, or it can be an existing file object
+   to read from or write to.
+
+   The mode argument can be either ``'r'`` for reading (default), ``'w'`` for
+   overwriting, 'a' for appending, or ``'x'`` for exclusive creation. These can
+   equivalently be given as ``'rb'``, ``'wb'``, ``'ab'``, and ``'xb'`` respectively. You may
+   also open in text mode with ``'rt'``, ``'wt'``, ``'at'``, and ``'xt'`` respectively.
+
+   When opening a file for reading, the *options* argument can be a dictionary
+   providing advanced decompression parameters; see
+   :class:`DecompressionParameter` for detailed information about supported
+   parameters. The *zstd_dict* argument is a :class:`ZstdDict` instance to be
+   used during decompression. When opening a file for reading, the *level*
+   argument should not be used.
```

Comment:
This may end up being an implementation question: if it "should not be used", does that mean it's ignored for reading (if so, perhaps say so)? Or does it cause problems if specified?

Reply:
Yeah, I think it would be good to say that it will raise an exception if passed during decompression.

Reply:
I re-worded this a little to call out that a TypeError is raised.